lib/swagger/swagger.ex

```diff
@@ -1,129 +1,57 @@
 defmodule Xcribe.Swagger do
   @moduledoc """
-  Treats list of Requests and generates OpenAPI 3.0 JSON
+  Treats list of Requests and generates OpenAPI Document with json format
   """
 
-  alias Xcribe.{Config, JSON}
-  alias Xcribe.Swagger.{Descriptor, Formatter}
+  alias Xcribe.{Config, JSON, Request}
+  alias Xcribe.Swagger.{Formatter}
 
-  @empty_data [nil, %{}, []]
-  @security_scheme %{
-    "securitySchemes" => %{
-      "api_key" => %{"name" => "Authorization", "type" => "apiKey", "in" => "header"}
-    }
-  }
+  import Xcribe.Swagger.Formatter, only: [raw_openapi_object: 0]
 
+  @doc """
+  Return an OpenAPI Document builted from the given requests list
+  """
   def generate_doc(requests) do
-    requests
-    |> build_openapi_object()
-    |> include_security_scheme(requests)
-    |> json_encode!()
+    raw_openapi_object()
+    |> mount_data_in_raw_object(requests)
+    |> json_encode()
```

Comment:
Just to be sure: is there any treatment on the possible error here? Why chaning from exception to `{:error, msg}`?

Reply:
I'm not sure if I understood you doubt. The only possible return for the pipe is a map. So there isn't a expected error. The wrapped function `JSON.encode!` was a suggestion the last PR to improve the function readability. Does it make sense?

Reply:
I mean to ask why was this private function renamed from `json_encode!` to `json_encode` (without the trailing `!`). It seems to me that `json_encode!` makes more sense as this version indicates that it may throw an exception.

Reply:
I agree. Fixed  =D